src/main/java/nextstep/qna/service/QnAService.java

```diff
@@ -26,24 +24,8 @@ public class QnAService {
     @Transactional
     public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
         Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
+        question.delete(loginUser);
+        List<DeleteHistory> deleteHistories = question.toDeleteHistories();
```

Comment:
질문을 삭제하는 시점에 삭제 상태를 `true`로 변경하는 것과 삭제 이력 반환을 동시에 처리해보면 어떨까요?
```suggestion
        List<DeleteHistory> deleteHistories = question.delete(loginUser);
```

Reply:
감사합니다. 이 코드가 두가지 역할을 한다고 생각해서, 분리 해보았는데요.
보통 Question 객체에 삭제 메시지를 보내고, 삭제 히스토리를 추가 호출하는 것보다
Question 객체에 삭제 메시지를 보내면, 삭제 히스토리를 반환해주는 편일까요?
리뷰어님은 어떻게 생각하시나요?

Reply:
지금처럼 두 메서드로 분리되어 있는데 만약 도메인에 대한 이해가 부족한 상태라면, 질문 똔느 답변을 삭제(`delete()`)하지도 않았는데 실수로 삭제 이력을 생성(`toDeleteHistories()`)할 가능성도 있기 때문에 후자의 구조가 조금 더 안전한 코드라고 생각합니다.